D3D12HelloWorld/Graphics/Rendering.cs

```diff
@@ -16,19 +16,27 @@ public abstract class ResourceView {
     }
 
     public class ConstantBufferView : ResourceView {
-        public ConstantBufferView(GraphicsResource resource) : base(resource, CreateConstantBufferView(resource)) {
+        public ConstantBufferView(ConstantBufferView constantBufferView) : base(constantBufferView.Resource, constantBufferView.CpuDescriptorHandle) {
         }
 
-        public ConstantBufferView(ConstantBufferView constantBufferView) : base(constantBufferView.Resource, constantBufferView.CpuDescriptorHandle) {
+        public ConstantBufferView(GraphicsResource resource)
+            : this(resource, 0, resource.SizeInBytes, resource.GraphicsDevice.ShaderResourceViewAllocator) {
         }
 
-        private static CpuDescriptorHandle CreateConstantBufferView(GraphicsResource resource) {
-            CpuDescriptorHandle cpuHandle = resource.GraphicsDevice.ShaderResourceViewAllocator.Allocate(1);
+        internal ConstantBufferView(GraphicsResource resource, ulong offset, ulong sizeInBytes, DescriptorAllocator shaderResourceViewAllocator)
+            : base(resource, CreateConstantBufferView(resource, offset, sizeInBytes, shaderResourceViewAllocator)) {
+        }
+
+        public static ConstantBufferView FromOffset<TBuffer>(GraphicsResource resource, ulong offset, DescriptorAllocator shaderResourceViewAllocator)
+            => new ConstantBufferView(resource, offset, (ulong)Unsafe.SizeOf<TBuffer>(), shaderResourceViewAllocator);
 
-            int constantBufferSize = ((int)resource.SizeInBytes + 255) & ~255;
+        private static CpuDescriptorHandle CreateConstantBufferView(GraphicsResource resource, ulong offset, ulong sizeInBytes, DescriptorAllocator shaderResourceViewAllocator) {
```

Comment:
Added new parameters `ulong offset, ulong sizeInBytes, DescriptorAllocator shaderResourceViewAllocator)` that are passed by the modified and new constructors to influence how this is created.

We now enable creating multiple views of the same resource at different offsets, a technique used by D3D12Bundles.